packages/skill-api/src/core/services/stripe-checkout.ts

```diff
@@ -313,9 +356,7 @@ export async function stripeCheckout({
             redeem_by: TWELVE_FOUR_HOURS_FROM_NOW,
             currency: 'USD',
             applies_to: {
-              products: [
-                loadedProduct.merchantProducts?.[0].identifier as string,
-              ],
+              products: [merchantProductIdentifier],
```

Comment:
No need to type cast to string here since we ensure it is a string when parsed earlier